React/wbooks/src/BookDetail/index.js

```diff
@@ -1,37 +1,55 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, { Component } from 'react';
+import { Link, withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
 
 import { BOOKS } from '../books_json';
+import { bookDetailAction } from '../redux/bookDetail/actions';
 
 import BookDetailInfo from './Components/BookDetailInfo';
 import BookSuggestions from './Components/BookSuggestions';
 import CommentSection from './Components/CommentSection';
 
-function BookDetail({ match }) {
-  const bookId = match.params.id;
-  const book = BOOKS.find(b => b.id == bookId);
-  const suggestions = BOOKS.filter(b => b.genre == book.genre);
+class BookDetail extends Component {
+  componentWillMount() {
+    this.props.dispatch(bookDetailAction.bookDetail(this.props.match.params.id));
+  }
+  render() {
+    const bookId = this.props.match.params.id;
+    const book = this.props.items && this.props.items.data ? this.props.items.data : null;
+    const suggestions = book ? (
+      <BookSuggestions suggestions={BOOKS.filter(b => b.genre == book.genre)} />
+    ) : null;
 
-  return (
-    <div>
-      <div className="body-container">
-        <Link to="/" className="back-button">
-          &lt; Volver
-        </Link>
-        <BookDetailInfo
-          imageUrl={book.image_url}
-          title={book.title}
-          author={book.author}
-          year={book.year}
-          genre={book.genre}
-        />
-        <hr />
-        <BookSuggestions suggestions={suggestions} />
-        <hr />
-        <CommentSection />
+    const bookComponent = book ? (
```

Comment:
This case is simple because there are only 2 components you want to hide if the request hasn't returned. But if you had more than that, you would have to make alot of constants asking if book exists. So instead, you could just:

```
return (
  ...
  { book && 
    <BookDetailInfo /*props*/ />
    <hr />
    <BookSuggestions /* props */ />
  }
  ...
)
```